test/support/single_cov.rb

```diff
@@ -0,0 +1,67 @@
+class SingleCov
+  COVERAGES = []
+
+  class << self
+    def expect(file, percent)
+      COVERAGES << [file, percent]
+    end
+
+    def verify!(result)
+      COVERAGES.each do |file, expected_percent|
+        uncovered!(file) unless coverage = result[File.expand_path(file)]
+
+        line_count = coverage.count { |l| l }
+        covered_lines = coverage.count { |l| l && l > 0 }
+        actual_percent = 100 * covered_lines / line_count
+        next if actual_percent == expected_percent
+
+        details = "#{actual_percent}% vs #{expected_percent}%"
+        if actual_percent > expected_percent
+          warn "#{file} exceeds expected coverage #{details}, increment expected coverage?"
+        else
+          warn "#{file} lower then expected coverage #{details}"
+          warn "Lines missing coverage:"
+          warn coverage.select { |c| c == 0 }.each_with_index.map { |c, i| "#{file}:#{i+1}" }
+          exit 1
```

Comment:
yep

On Mon, Apr 11, 2016 at 9:33 PM, Roman Sandler <notifications@github.com>
wrote:

> In test/support/single_cov.rb
> <https://github.com/zendesk/samson/pull/805#discussion_r59319595>:
>
> > +      COVERAGES.each do |file, expected_percent|
> > +        uncovered!(file) unless coverage = result[File.expand_path(file)]
> > +
> > +        line_count = coverage.count { |l| l }
> > +        covered_lines = coverage.count { |l| l && l > 0 }
> > +        actual_percent = 100 * covered_lines / line_count
> > +        next if actual_percent == expected_percent
> > +
> > +        details = "#{actual_percent}% vs #{expected_percent}%"
> > +        if actual_percent > expected_percent
> > +          warn "#{file} exceeds expected coverage #{details}, increment expected coverage?"
> > +        else
> > +          warn "#{file} lower then expected coverage #{details}"
> > +          warn "Lines missing coverage:"
> > +          warn coverage.select { |c| c == 0 }.each_with_index.map { |c, i| "#{file}:#{i+1}" }
> > +          exit 1
>
> this doesn't actually kill the process immediately because everything is
> running at_exit? It just makes sure the code is not 0?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/805/files/f77d1b4a4260626d6b493ac6ea4f4aaa0cf1fc7e#r59319595>
>